crates/topos-tce-transport/src/lib.rs

```diff
@@ -101,14 +114,80 @@ pub enum ProtocolEvents {
     /// Indicates that 'echo' message broadcasting is required
     Echo {
         certificate_id: CertificateId,
+        signature: Signature,
+        validator_id: ValidatorId,
     },
     /// Indicates that 'ready' message broadcasting is required
     Ready {
         certificate_id: CertificateId,
+        signature: Signature,
+        validator_id: ValidatorId,
     },
     /// For simulation purpose, for now only caused by ill-formed sampling
     Die,
 
     /// Stable Sample
     StableSample,
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum Error {}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq, Hash)]
+pub struct ValidatorId(H160);
+
+impl ValidatorId {
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+
+    pub fn address(&self) -> Address {
+        Address::from_str(&format!("0x{}", hex::encode(self.0))).unwrap()
```

Comment:
if `Address` is `ethers::types::Address` then this method is somehow useless as `Address` is an alias to `H160`

Reply:
It's just a convience function to not to have to to `validator_id.0`.